StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1407SA1408FixAllProvider.cs

```diff
@@ -25,13 +24,10 @@ internal sealed class SA1407SA1408FixAllProvider : DocumentBasedFixAllProvider
                 return null;
             }
 
-            var newDocument = document;
+            var root = await document.GetSyntaxRootAsync(fixAllContext.CancellationToken).ConfigureAwait(false);
 
-            var root = await newDocument.GetSyntaxRootAsync(fixAllContext.CancellationToken).ConfigureAwait(false);
+            List<SyntaxNode> nodes = new List<SyntaxNode>();
 
-            // First annotate all expressions that need parenthesis with a temporary annotation.
-            // With this annotation we can find the nodes that need parenthesis even if
-            // the source span changes.
             foreach (var diagnostic in diagnostics)
             {
                 SyntaxNode node = root.FindNode(diagnostic.Location.SourceSpan);
```

Comment:
:thought_balloon: We should investigate whether this needs `getInnermostNodeForTie: true`.